compiler/src/iree/compiler/Dialect/LinalgExt/IR/TilingInterfaceImpl.cpp

```diff
@@ -1440,6 +1440,222 @@ LogicalResult ArgCompareOp::generateScalarImplementation(OpBuilder &b,
   return success();
 }
 
+FailureOr<SmallVector<Value>>
+ArgCompareOp::generateInitialTensorForPartialReduction(
+    OpBuilder &b, Location loc, ArrayRef<OpFoldResult> sizes,
+    const llvm::SetVector<unsigned> &reductionDims) {
+  // Get the original init tensors.
+  Value valueInit = outputValue();
+  Value indexInit = outputIndex();
+
+  // Create tensors with the partial result shape.
+  Type valueElTy = getElementTypeOrSelf(valueInit.getType());
+  Type indexElTy = getElementTypeOrSelf(indexInit.getType());
+  SmallVector<OpFoldResult> partialResultShape(sizes);
+  Value emptyValueTensor =
+      tensor::EmptyOp::create(b, loc, partialResultShape, valueElTy);
+  Value emptyIndexTensor =
+      tensor::EmptyOp::create(b, loc, partialResultShape, indexElTy);
+
+  // Broadcast init values to partial result shape for slicing inside
+  // scf.forall. Each tile in the parallel loop will extract slices from
+  // these broadcasted tensors to get initialized values for the ArgCompareOp.
+  // Example: tensor<64xf32> -> tensor<64x32xf32> for 32 reduction tiles along
+  // dim 1.
+  SmallVector<int64_t> broadcastDims =
+      llvm::to_vector_of<int64_t>(reductionDims);
```

Comment:
```suggestion
  auto broadcastDims =
      llvm::to_vector_of<int64_t>(reductionDims);
```
See https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable